core/src/CMakeLists.txt

```diff
@@ -113,6 +112,34 @@ IF (Kokkos_ENABLE_IMPL_DESUL_ATOMICS AND NOT desul_FOUND)
   )
 ENDIF()
 
+IF (KOKKOS_ENABLE_IMPL_MDSPAN)
+  MESSAGE(STATUS "Experimental mdspan support is enabled")
+  if (KOKKOS_ENABLE_MDSPAN_EXTERNAL)
+    MESSAGE(STATUS "Using external mdspan")
+    target_link_libraries(kokkoscore PUBLIC std::mdspan)
+  else()
+    KOKKOS_LIB_INCLUDE_DIRECTORIES(
+      kokkoscore
+      ${PROJECT_SOURCE_DIR}/tpls/mdspan/include
```

Comment:
Not sure that this works with trilinos, I think there is a reason I did the CMAKE_CURRENT_SOURCE_DIR above